### ISSUE TITLE
Create setting/filter to stop saving of submissions

### ISSUE THREAD

**Issue by sparkalow** [libreform/wp-libre-form#95]
feature request:
Some use cases do not require storing of data in the DB and with the upcoming [GDPR](https://en.wikipedia.org/wiki/General_Data_Protection_Regulation) (25 May 2018) it could be beneficial not to store personal information

**Comment by k1sul1** [comment]
We've discussed about a similar feature, maybe even a "delete my data" button sent to the submitter via email. 

Will implement something like this in the near future.

**Comment by k1sul1** [comment]
There's also an easy "contributor" badge up for grabs, we'd gladly accept a PR preventing saving. I'm pretty sure you could implement it already with WP hooks (save_post, get post type from id, and so on...), but supporting it directly could be good.

I'd implement this with two filters: 
```
wplf-form-disable-save
wplf-form-{id}-disable-save
```

with the id obviously taking precedence.

**Comment by k1sul1** [comment]
If someone doesn't claim this, I'm probably taking this today.

**Comment by k1sul1** [comment]
Turns out this might be a bit of a feature creep, because of the other features we have. I'll use file uploads as an example. 

File uploads are always "assigned" to a form submission. Not saving form submissions to the database is not necessarily a problem directly, but if you have a file upload field in a form that won't save it's submissions to the database, then you might end up with hundreds or thousands of files in the media library, with no trail whatsoever associated with them. I won't go into detail on how this could be used maliciously, but it could be. I wouldn't want to accidentally host CP as a site owner.

This could obviously be worked around, but I'd have to implement a number of things, like associate form metadata to the uploaded file, which is only possible for items in the media library, so that's not good enough. To support non-media library uploads as well, something like a wplf-upload CPT would have to be created, and a load of code would have to be written.

That increases the complexity of our codebase more than I'd like. Nothing too serious, but I'd also like to question the necessity of this feature's existence. Anyone who's ever used a form plugin that doesn't save form submissions to the database, but only sends them via email has probably experienced issues like mail going to spam, or mail not being sent at all. Not very nice when that happens.

If an user fills a form with personal information, it's expected that there's a reason for gathering that data. GDPR wise, it's totally fine to store that data in the database. Unless you're only sending that data to the user that submitted it (which doesn't make any sense), email isn't any better, you're fucked nevertheless.  

I'm still up for implementing a "delete your own data" button in submission confirmation email, but that's something that would require a dedicated setting on each form. I don't want to add that in 2.0, as 2.0 already has a lot of new features, and I want to get it out sooner than later. Maybe 2.1 will have that feature.

All hope is not lost though. For 2.0, I might just implement a required capability setting for viewing form submissions so that anyone who has an Editor level account on the site isn't a GDPR data processor. That alone should be enough to be GDPR-friendly.

*TL;DR*: This isn't happening in the near future. If you don't want to keep form submissions in the database, you can remove them pretty easily immediately after form handling is complete: 
```php
add_action("wplf_123_post_validate_submission", function($data) {
  wp_delete_post($data->submission_id, true);
});
```
